homeassistant/components/vevor_heater/config_flow.py

```diff
@@ -0,0 +1,123 @@
+"""Config flow for Vevor Heater integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from home_assistant_bluetooth import BluetoothServiceInfoBleak
+from vevor_heater_ble.heater import VevorDevice
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth import async_discovered_service_info
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("address"): str,
+    }
+)
+
+
+async def validate_device(hass: HomeAssistant, address: str) -> dict[str, Any]:
+    """Validate the given device."""
+    if not address:
```

Comment:
This looks unreachable through the normal flow

Reply:
AIUI and as my experiments show this gets called -  at the BT discovery step (`async_step_bluetooth`). Given the entire integration is reverse engineered and there's no documentation for the device I think actually communicating with the device like we do here is the only way to find out if it is what we think it is. 

I'm of course happy to take advice on the matter as my experience with HA design principles and best practices is... limited.

Reply:
address is typed to `str` so if this was reachable it would be a bug in the bluetooth stack. We should take it out